aptos-move/aptos-aggregator/src/delta_change_set.rs

```diff
@@ -213,53 +175,27 @@ impl DeltaOp {
     }
 }
 
-/// Implements application of `Addition` to `base`.
-pub fn addition(base: u128, value: u128, limit: u128) -> PartialVMResult<u128> {
-    if limit < base || value > (limit - base) {
-        Err(abort_error(
-            format!("overflow when adding {} to {}", value, base),
-            EADD_OVERFLOW,
-        ))
-    } else {
-        Ok(base + value)
-    }
-}
-
-/// Implements application of `Subtraction` to `base`.
-pub fn subtraction(base: u128, value: u128) -> PartialVMResult<u128> {
-    if value > base {
-        Err(abort_error(
-            format!("underflow when subtracting {} from {}", value, base),
-            ESUB_UNDERFLOW,
-        ))
-    } else {
-        Ok(base - value)
-    }
-}
-
-/// Error for delta application. Can be used by delta partial functions
-/// to return descriptive error messages and an appropriate error code.
-fn abort_error(message: impl ToString, code: u64) -> PartialVMError {
-    PartialVMError::new(StatusCode::ABORTED)
-        .with_message(message.to_string())
-        .with_sub_status(code)
-}
-
 impl std::fmt::Debug for DeltaOp {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self.update {
-            DeltaUpdate::Plus(value) => {
+            DeltaValue::Positive(value) => {
                 write!(
                     f,
                     "+{} ensures 0 <= result <= {}, range [-{}, {}]",
-                    value, self.limit, self.min_negative, self.max_positive
+                    value,
+                    self.max_value,
+                    self.history.min_achieved_negative_delta,
```

Comment:
Let's implement Debug for history, and {:?} history directly here, also will make sure it shows the new fields.